nfr-builder: no rule to make target `dfu_private.pem`

Full error:

```
make: *** No rule to make target '../test-certificates/firmware-nrf52/dfu_private.pem', needed by 'develop-nk3am-nrf52-1.5.0-test.20230704.zip'.  Stop.
```

It looks like the file was renamed in #338 but not in the makefile.

--- a/utils/nrf-builder/Makefile
+++ b/utils/nrf-builder/Makefile
@@ -10,7 +10,7 @@ NK3_BL_DIR := $(NK3_FW_DIR)/utils/nrf-bootloader
 NK3_BL_KEY_DIR := $(NK3_FW_DIR)/utils/nrf-bootloader/signing-key
 
 KEY_DIR := $(TEST_CERTS_DIR)/firmware-nrf52
-PRIV_SIGN_KEY := $(KEY_DIR)/dfu_private.pem
+PRIV_SIGN_KEY := $(KEY_DIR)/dfu_private.key
 
 BL_OUTPUT := mbr.hex bootloader.hex
 

